tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -653,56 +660,62 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 		err = oa.WaitForTidbClusterReady(tc, 10*time.Minute, 5*time.Second)
 		framework.ExpectNoError(err, "Expected get tidbcluster")
 
+		ginkgo.By("deploy tidbmonitor")
 		tm := fixture.NewTidbMonitor("monitor-test", ns, tc, true, true, false)
-		deletePVP := corev1.PersistentVolumeReclaimDelete
-		tm.Spec.PVReclaimPolicy = &deletePVP
+		pvpDelete := corev1.PersistentVolumeReclaimDelete
+		tm.Spec.PVReclaimPolicy = &pvpDelete
 		_, err = cli.PingcapV1alpha1().TidbMonitors(ns).Create(tm)
 		framework.ExpectNoError(err, "Expected tidbmonitor deployed success")
 		err = tests.CheckTidbMonitor(tm, cli, c, fw)
 		framework.ExpectNoError(err, "Expected tidbmonitor checked success")
-		pvc, err := c.CoreV1().PersistentVolumeClaims(ns).Get(monitor.GetMonitorFirstPVCName(tm.Name), metav1.GetOptions{})
-		framework.ExpectNoError(err, "Expected fetch tidbmonitor pvc success")
 
-		pvName := pvc.Spec.VolumeName
-		pv, err := c.CoreV1().PersistentVolumes().Get(pvName, metav1.GetOptions{})
-		framework.ExpectNoError(err, "Expected fetch tidbmonitor pv success")
+		ginkgo.By("check tidbmonitor pv label")
+		err = wait.Poll(5*time.Second, 2*time.Minute, func() (done bool, err error) {
+			pvc, err := c.CoreV1().PersistentVolumeClaims(ns).Get(monitor.GetMonitorFirstPVCName(tm.Name), metav1.GetOptions{})
+			framework.ExpectNoError(err, "Expected fetch tidbmonitor pvc success")
+			pvName := pvc.Spec.VolumeName
+			pv, err := c.CoreV1().PersistentVolumes().Get(pvName, metav1.GetOptions{})
+			framework.ExpectNoError(err, "Expected fetch tidbmonitor pv success")
```

Comment:
Why do you put these lines into this `Poll` but still `ExpectNoError` (rather than return `false`)?

Reply:
Good one!
The original `Poll` is meaning less because nothing has changed between each poll. But I forgot to change the `ExpectNoError` :frowning: